plugins/CoreAdminHome/OptOutManager.php

```diff
@@ -205,6 +205,20 @@ public function getOptOutJSEmbedCode(
         bool $applyStyling,
         bool $showIntro
     ): string {
+        $parsedUrl = parse_url($matomoUrl);
+
+        if (
+            (!empty($matomoUrl) && false === $parsedUrl)
+            || (!empty($parsedUrl['scheme']) && !in_array(strtolower($parsedUrl['scheme']), ['http', 'https']))
+            || (empty($parsedUrl['host']) || !Url::isValidHost($parsedUrl['host']))
+        ) {
+            throw new \Piwik\Exception\Exception('The provided URL is invalid.');
+        }
+
+        // We put together the url based on the parsed parameters manually to ensure it might not include unexpected values
+        // for protocol less urls starting with //, we need to prepend the double slash again
+        $matomoUrl = (strpos($matomoUrl, '//') === 0 ? '//' : '') . UrlHelper::getParseUrlReverse($parsedUrl);
```

Comment:
```suggestion
        $matomoUrl = (strpos($matomoUrl, '//') === 0 ? '//' : '') . \Piwik\UrlHelper::getParseUrlReverse($parsedUrl);
```

Or we add the appropriate `use` statement at the top. Otherwise `PrivacyManager_spec` fails with an exception.